config.go

```diff
@@ -118,13 +118,10 @@ type Config struct {
 
 	LoggerFactory logging.LoggerFactory
 
-	// ConnectContextMaker is a function to make a context used in Dial(),
-	// Client(), Server(), and Accept(). If nil, the default ConnectContextMaker
-	// is used. It can be implemented as following.
+	// ConnectContextMaker is no longer used. It is kept for compatibility
+	// reasons and will be removed in a future release.
 	//
-	// 	func ConnectContextMaker() (context.Context, func()) {
-	// 		return context.WithTimeout(context.Background(), 30*time.Second)
-	// 	}
+	// Deprecated: Use the context parameter in [HandshakeContext] instead.
```

Comment:
I think we should kill this now. We are going with a major release, lets break now :)